tests/lib/rules/no-array-prototype-extensions.js

```diff
@@ -550,10 +550,25 @@ import { get as g } from 'dummy';
       errors: [{ messageId: 'main', type: 'CallExpression' }],
     },
     {
+      // When unexpected number of arguments are passed, auto-fixer will not run
       code: 'something.reject()',
       output: null,
       errors: [{ messageId: 'main', type: 'CallExpression' }],
     },
+    {
+      // Single argument is passed
+      code: 'something.reject(function(el) { return el.isFlagged; })',
+      output:
+        'something.filter(function() { return !(function(el) { return el.isFlagged; }).apply(this, arguments); })',
```

Comment:
Can we tweak this autofix (ignore the spacing, concise one-line is good):

```js
something.filter(function (...args) {
  return !Reflect.apply(
    function (el) {
      return el.isFlagged;
    },
    this,
    args
  );
});
```

This addresses violations from two lint rules:

* [unicorn/prefer-reflect-apply](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v42.0.0/docs/rules/prefer-reflect-apply.md)
* [prefer-rest-params](https://eslint.org/docs/latest/rules/prefer-rest-params)

Reply:
@bmish Yup should be doable. However, `Reflect` is not supported on IE11. Do you think it is safe to use it?

Reply:
I'm not too worried about IE11 but if we want to be safe, we could add a TODO comment to switch to Reflect, the same way we did for [includes](https://github.com/ember-cli/eslint-plugin-ember/pull/1617/files#r990294557):

```
// TODO: Switch to `Reflect` once Ember v3 LTS support ends: https://emberjs.com/releases/lts/
```

Either way, let's fix the prefer-rest-params issue.

Reply:
Sure. Will add a TODO just to be safe.

Reply:
Fixed it. Can you take a relook?